swagger2.0/oic.r.rule.swagger.json

```diff
@@ -0,0 +1,371 @@
+              "p":  {"bm": 3},
+              "rep": {
+                "rule": "(timeperiod:triggered = true)",
+                "ruleresult": false,
+                "ruleenable": false,
+                "actionenable": false
+              }
+            },
+            "schema": { "$ref": "#/definitions/rule-create" }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "interface-all": {
+      "in": "query",
+      "name": "if",
+      "type": "string",
+      "enum": ["oic.if.ll","oic.if.baseline","oic.if.create"]
+    }
+  },
+  "definitions": {
+    "rule-create": {
+      "type": "object",
+      "properties": {
+        "href": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/href"
+        },
+        "if": {
+          "description": "The OCF Interfaces supported by the created Resource",
+          "items": {
+            "enum": [
+              "oic.if.baseline",
+              "oic.if.ll",
+              "oic.if.b",
+              "oic.if.rw",
+              "oic.if.r",
+              "oic.if.a",
+              "oic.if.s"
+            ],
+            "type": "string",
+            "maxLength": 64
+          },
+          "minItems": 1,
+          "uniqueItems": true,
+          "type": "array"
+        },
+        "ins": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/ins"
+        },
+        "p": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/p"
+        },
+        "rt": {
+          "description": "Resource Type of the created Resource",
+          "items": {
+            "maxLength": 64,
+            "type": "string",
+            "enum": ["oic.r.rule.inputcollection","oic.r.rule.expression","oic.r.rule.actioncollection"]
+          },
+          "minItems": 1,
+          "uniqueItems": true,
+          "type": "array"
+        },
+        "rep": {
+          "type": "object"
+        }
+      }
+    },
+    "oic.oic-link": {
+      "type": "object",
+      "properties": {
+        "anchor": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/anchor"
+        },
+        "di": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/di"
+        },
+        "eps": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/eps"
+        },
+        "href": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.links.properties.core-schema.json#/definitions/href"
+        },
+        "if": {
+          "description": "The OCF Interfaces supported by the Linked Resource",
+          "items": {
+            "enum": [
```

Comment:
All of these OCF Interfaces aren't required.  Only the ones exposed by the Resource Types that may be included in the Collection.

Reply:
No create on rule action collection, but yes to linkadd later.

Reply:
Do you also agree that we can remove the OCF Interfaces that aren't exposed by one of the Resource Types that go into the Rule Collection (e.g. oic.if.r, oic.if.s, etc.).
As discussed elsewhere, oic.if.create may still need to be added here.